### DISCUSSION TITLE
[Question] Import Objaverse glb file into Isaac Sim

### DISCUSSION THREAD

**Discussion by supersglzc** [isaac-sim/IsaacLab#595] [Q&A]
### Question

Hi, I would like to import some objects from Objaverse. They are glb files and I want to convert them to usd files. Is there any exist converter for this? 

I found the asset converter https://docs.omniverse.nvidia.com/extensions/latest/ext_asset-converter.html in Omniverse but I got no such module error. 

Thanks!

**Comment by Mayankm96** [comment]
We have some instructions on importing new assets: https://isaac-sim.github.io/IsaacLab/source/how-to/import_new_asset.html#using-mesh-importer

Hope that helps!

**Comment by supersglzc** [comment]
Thanks for the reply! This importer does convert the mesh to usd. However, the collision mesh does not exist and it only has a geometry Xform, even though I select the collision approximation to convexDecomposition. 

I knew that the mesh itself has collider property, but when I import it into my environment, it gives me very weird collision.

**Reply by Mayankm96** [reply]
The collider mesh is applied directly to the visual mesh itself. However, this is currently limited to convex decomposition, while PhysX supports many more collider approximations, such as convex hull, SDF, etc.

Is the weird collision that you see an artifact of the collision approximation?

**Reply by supersglzc** [reply]
I think the problem is due to the instanceable. If I set make instanceable to true, the collision is weird and the object can penetrate into other objects. If I disable the instanceable flag, then everything looks good. But it is important for me to use instanceable objects. Do you have any idea? Thanks!

**Reply by aditya-nisal** [reply]
Hello @supersglzc @Mayankm96 , were you able to see the materials and textures for all the objaverse objects in isaac sim after the conversion? I am able to convert the .glb to .usd, but I am missing the textures and details. Any script or repo you found useful to resolve this error? Any help would be really appreciated. Thanks
